src/zope/schema/_bootstrapfields.py

```diff
@@ -529,8 +532,15 @@ def fromUnicode(self, str):
         ...
         zope.schema._bootstrapinterfaces.ConstraintNotSatisfied: (u'foo spam', '')
         """
-        self.validate(str)
-        return str
+        try:
```

Comment:
This could be simplified by using `from zope.schema._compat import text_type` at the top of the module.